{
  "id": "RHSA-2025:10140",
  "summary": "Red Hat Security Advisory: python3.12 security update",
  "modified": "2025-07-02T10:44:17.453228Z",
  "published": "2025-07-02T10:03:37Z",
  "upstream": [
    "CVE-2024-12718",
    "CVE-2025-4138",
    "CVE-2025-4330",
    "CVE-2025-4435",
    "CVE-2025-4517"
  ],
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://access.redhat.com/errata/RHSA-2025:10140"
    },
    {
      "type": "ARTICLE",
      "url": "https://access.redhat.com/security/updates/classification/#important"
    },
    {
      "type": "REPORT",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2370010"
    },
    {
      "type": "REPORT",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2370013"
    },
    {
      "type": "REPORT",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2370014"
    },
    {
      "type": "REPORT",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2370016"
    },
    {
      "type": "REPORT",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2372426"
    },
    {
      "type": "ADVISORY",
      "url": "https://security.access.redhat.com/data/csaf/v2/advisories/2025/rhsa-2025_10140.json"
    },
    {
      "type": "REPORT",
      "url": "https://access.redhat.com/security/cve/CVE-2024-12718"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.cve.org/CVERecord?id=CVE-2024-12718"
    },
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-12718"
    },
    {
      "type": "ARTICLE",
      "url": "https://gist.github.com/sethmlarson/52398e33eff261329a0180ac1d54f42f"
    },
    {
      "type": "ARTICLE",
      "url": "https://github.com/python/cpython/commit/3612d8f51741b11f36f8fb0494d79086bac9390a"
    },
    {
      "type": "ARTICLE",
      "url": "https://github.com/python/cpython/commit/9e0ac76d96cf80b49055f6d6b9a6763fb9215c2a"
    },
    {
      "type": "ARTICLE",
      "url": "https://github.com/python/cpython/issues/127987"
    },
    {
      "type": "ARTICLE",
      "url": "https://github.com/python/cpython/issues/135034"
    },
    {
      "type": "ARTICLE",
      "url": "https://github.com/python/cpython/pull/135037"
    },
    {
      "type": "ARTICLE",
      "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/MAXIJJCUUMCL7ATZNDVEGGHUMQMUUKLG/"
    },
    {
      "type": "REPORT",
      "url": "https://access.redhat.com/security/cve/CVE-2025-4138"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.cve.org/CVERecord?id=CVE-2025-4138"
    },
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2025-4138"
    },
    {
      "type": "REPORT",
      "url": "https://access.redhat.com/security/cve/CVE-2025-4330"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.cve.org/CVERecord?id=CVE-2025-4330"
    },
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2025-4330"
    },
    {
      "type": "REPORT",
      "url": "https://access.redhat.com/security/cve/CVE-2025-4435"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.cve.org/CVERecord?id=CVE-2025-4435"
    },
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2025-4435"
    },
    {
      "type": "REPORT",
      "url": "https://access.redhat.com/security/cve/CVE-2025-4517"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.cve.org/CVERecord?id=CVE-2025-4517"
    },
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2025-4517"
    }
  ],
  "affected": [
    {
      "package": {
        "name": "python-unversioned-command",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python-unversioned-command"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3-debug",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3-debug"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3-devel",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3-devel"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3-idle",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3-idle"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3-libs",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3-libs"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3-test",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3-test"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3-tkinter",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3-tkinter"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3.12",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3.12"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3.12-debuginfo",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3.12-debuginfo"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    },
    {
      "package": {
        "name": "python3.12-debugsource",
        "ecosystem": "Red Hat:enterprise_linux:10.0",
        "purl": "pkg:rpm/redhat/python3.12-debugsource"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0:3.12.9-2.el10_0.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://security.access.redhat.com/data/osv/RHSA-2025:10140.json"
      },
      "versions": []
    }
  ],
  "schema_version": "1.6.0",
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:H/I:H/A:L"
    }
  ]
}